### DISCUSSION TITLE
[Q&A]Expanding the number of drives in a distributed deployment

### DISCUSSION THREAD

**Discussion by A-BenMao** [minio/minio#20291] [Q&A] [answered]
## Expected Behavior
First of all, thank the minio project

We have a minio consisting of 5 machines. Currently, we only use 16 of the 24 disk slots. We want to use the remaining 8 disk slots as well.

Changing the MINIO_VOLUMES parameter to include the new drive should expand the number of drives in the pool:

Original: MINIO_VOLUMES="http://10.10.208.{1...5}:9000/data/minio-sd{1...16}"
Modified: MINIO_VOLUMES="http://10.10.208.{1...5}:9000/data/minio-sd{1...16}   http://10.10.208.{1...5}:9000/data/minio-sd{17...24}"

## Current Behavior
This is just the expected idea. From the documentation, my understanding is that you cannot expand the disk on the same machine.

## Possible Solution
Can it be expanded as I described? Thanks

## Steps to Reproduce (for bugs)
1.

## Your Environment

* Version used (`minio --version`):
minio version RELEASE.2024-06-26T01-06-18Z (commit-id=499531f0b5863ad5b3a5e92a24239138c3ba7a13)
Runtime: go1.22.4 linux/amd64
License: GNU AGPLv3 - https://www.gnu.org/licenses/agpl-3.0.html
Copyright: 2015-2024 MinIO, Inc.

* Server setup and configuration:
MINIO_VOLUMES="http://10.10.208.{1...5}:9000/data/minio-sd{1...16}"
* Operating System and version (`uname -a`):
Linux bm-hz-stable-minio1 4.18.0-372.9.1.el8.x86_64  Tue May 10 14:48:47 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Comment by marktheunissen** [comment, marked as answer]
Hi @A-BenMao , it is not possible to add drives to an existing node and just restart MinIO with the new drives. We don't support that.

We support expansion by adding a new pool: https://min.io/docs/minio/linux/operations/install-deploy-manage/expand-minio-deployment.html#expand-minio-distributed

Another option you have is to move the data off the existing cluster (decommission) and then start again, and move the data back

Also see where it has been asked before: https://github.com/minio/minio/discussions/19280

**Reply by A-BenMao** [reply]
Thanks for your answer. That is to say, if we don't migrate the data, we can't use the free disk space[The 8 disks I mentioned above], right?

**Reply by marktheunissen** [reply]
Yes that's correct
